server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -967,24 +1004,82 @@ public void testExceptionOnResponseProcessing() throws Exception {
     }
 
+    public void testCatchExceptionOnRequestProcessing() {
+        SearchRequestProcessor throwingRequestProcessor = new FakeRequestProcessor("throwing_request", null, null, true, r -> {
+            throw new RuntimeException();
+        });
+        Map<String, Processor.Factory<SearchRequestProcessor>> throwingRequestProcessorFactory = Map.of(
+            "throwing_request",
+            (pf, t, i, f, c) -> throwingRequestProcessor
+        );
+
+        SearchPipelineService searchPipelineService = createWithProcessors(
+            throwingRequestProcessorFactory,
+            Collections.emptyMap(),
+            Collections.emptyMap()
+        );
+
+        Map<String, Object> pipelineSourceMap = new HashMap<>();
+        pipelineSourceMap.put(Pipeline.REQUEST_PROCESSORS_KEY, List.of(Map.of("throwing_request", Collections.emptyMap())));
+
+        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource().searchPipelineSource(pipelineSourceMap);
+        SearchRequest searchRequest = new SearchRequest().source(sourceBuilder);
+
+        // Caught Exception thrown when processing the request
+        PipelinedRequest pipelinedRequest = searchPipelineService.resolvePipeline(searchRequest);
+    }
+
+    public void testCatchExceptionOnResponseProcessing() throws Exception {
+        SearchResponseProcessor throwingResponseProcessor = new FakeResponseProcessor("throwing_response", null, null, true, r -> {
+            throw new RuntimeException();
+        });
+        Map<String, Processor.Factory<SearchResponseProcessor>> throwingResponseProcessorFactory = Map.of(
+            "throwing_response",
+            (pf, t, i, f, c) -> throwingResponseProcessor
+        );
+
+        SearchPipelineService searchPipelineService = createWithProcessors(
+            Collections.emptyMap(),
+            throwingResponseProcessorFactory,
+            Collections.emptyMap()
+        );
+
+        Map<String, Object> pipelineSourceMap = new HashMap<>();
+        pipelineSourceMap.put(Pipeline.RESPONSE_PROCESSORS_KEY, List.of(Map.of("throwing_response", Collections.emptyMap())));
+
+        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource().size(100).searchPipelineSource(pipelineSourceMap);
+        SearchRequest searchRequest = new SearchRequest().source(sourceBuilder);
+
+        PipelinedRequest pipelinedRequest = searchPipelineService.resolvePipeline(searchRequest);
+
+        // Caught Exception thrown when processing response
+        SearchResponse response = new SearchResponse(null, null, 0, 0, 0, 0, null, null);
+        pipelinedRequest = searchPipelineService.resolvePipeline(searchRequest);
+        SearchResponse transformedResponse = pipelinedRequest.transformResponse(response);
+
+        // No change to the response
+        assertSame(transformedResponse, response);
+
+    }
+
     public void testStats() throws Exception {
```

Comment:
It's probably worth including one test to confirm that the `failed` metrics get incremented when we ignore failures.

With the exception handling in `FakeRequestProcessor` and `FakeResponseProcessor`, I think such a test would have failed (since we wouldn't have called `onProcess{Request,Response}Failed`).

Reply:
Added testStatsEnabledIgnoreFailure and the count also increments with enabledIgnoreFailure.